src/annotator/anchoring/test/types-test.js

```diff
@@ -19,46 +21,335 @@ describe('types', function () {
+    }
+
+    describe('#fromRange', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        fakePosFromRange.returns({
+          start: 0,
+          end: 1,
+        });
+        const anchor = TextPositionAnchor.fromRange(container, new Range());
+        assert.calledOnce(fakePosFromRange);
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#fromSelector', () => {
+      it('returns a TextPositionAnchor instance', () => {
+        const anchor = TextPositionAnchor.fromSelector(container, {
+          start: 0,
+          end: 0,
+        });
+        assert.instanceOf(anchor, TextPositionAnchor);
+      });
+    });
+
+    describe('#toSelector', () => {
+      it('returns a TextPositionSelector', () => {
+        const anchor = createTextPositionAnchor();
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 0,
+        });
+      });
+    });
+
+    describe('#toRange', () => {
+      it('returns a range object', () => {
+        const anchor = createTextPositionAnchor();
+        fakePosToRange.returns('fake range');
+        assert.equal(anchor.toRange(), 'fake range');
+        assert.calledWith(fakePosToRange, container, { start: 0, end: 0 });
+      });
+    });
+
+    describe('integration tests', () => {
+      beforeEach(() => {
+        // restore dom-anchor-text-position to test third party lib integration
+        $imports.$restore();
+      });
+
+      it('can convert a Range to TextPositionSelector and back to a Range', () => {
+        const range = document.createRange();
+        range.setStart(container.firstChild, 0);
+        range.setEnd(container.firstChild, 4);
+        const anchor = TextPositionAnchor.fromRange(container, range);
+        assert.deepEqual(anchor.toSelector(), {
+          type: 'TextPositionSelector',
+          start: 0,
+          end: 4,
+        });
+        const newRange = anchor.toRange();
+        assert.deepEqual(range, newRange);
+        assert.equal(newRange.toString(), 'Four');
+      });
+    });
+  });
+
+  describe('TextQuoteAnchor', () => {
+    let fakeQuoteToRange;
+    let fakeQuoteFromRange;
+    let fakeToTextPosition;
+
+    beforeEach(() => {
+      fakeQuoteToRange = sinon.stub();
+      fakeQuoteFromRange = sinon.stub();
+      fakeToTextPosition = sinon.stub();
+      $imports.$mock({
+        'dom-anchor-text-quote': {
+          fromRange: fakeQuoteFromRange,
+          toRange: fakeQuoteToRange,
+          toTextPosition: fakeToTextPosition,
+        },
+      });
+    });
+
+    afterEach(() => {
+      $imports.$restore();
+    });
+
+    function createTextQuoteAnchor() {
+      return new TextQuoteAnchor(container, 'Liberty', {
+        prefix: 'a new nation, conceived in ',
+        suffix: ', and dedicated to the proposition that',
+      });
+    }
+
+    describe('#fromRange', () => {
+      it('returns a range object', () => {
```

Comment:
```suggestion
      it('returns a TextQuoteAnchor', () => {
```

`fromRange` takes a range as input rather than returning a range.